src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs

```diff
@@ -262,10 +262,12 @@ await subscriber.Value.InitializeSubscriberAsync(this, _activeConfiguredProjectS
 
                     foreach (var subscriber in Subscribers)
                     {
-                        subscriber.Value.AddSubscriptionsAsync(newProjectContext);
+                        Shell.ThreadHelper.JoinableTaskFactory.Run(() => subscriber.Value.AddSubscriptionsAsync(newProjectContext));
```

Comment:
you cannot await inside a lock in C# (for obvious reasons).

Its unclear to me what the correct thing to do here is.  Previously, we were in a lock but we did not wait for async operations to complete before leaving the lock.

either:
1. This code does not need to be in the lock, in which case we should move it out of the lock
2. This code does need to be in the lock, in which case we need to actually wait for the async operations to complete or we risk bugs from threading behavior.